tools/distpartitioning/dataset_utils.py

```diff
@@ -48,60 +48,171 @@ def get_dataset(input_dir, graph_name, rank, world_size, schema_map):
-            else:
-                node_features[ntype_name+'/'+feat_name] = torch.from_numpy(np.load(input_dir+my_feat_data[0]))
-
-            node_feature_tids[ntype_name].append([feat_name, my_feat_data[1], my_feat_data[2]])
+    dataset_features = schema_map[constants.STR_NODE_DATA]
+    if((dataset_features is not None) and (len(dataset_features) > 0)):
+        for ntype_name, ntype_feature_data in dataset_features.items():
+            #ntype_feature_data is a dictionary
+            #where key: feature_name, value: dictionary in which keys are "format", "data"
+            node_feature_tids[ntype_name] = []
+            for feat_name, feat_data in ntype_feature_data.items():
+                assert len(feat_data[constants.STR_DATA]) == world_size
+                assert feat_data[constants.STR_FORMAT][constants.STR_NAME] == constants.STR_NUMPY
+                my_feat_data_fname = feat_data[constants.STR_DATA][rank] #this will be just the file name
+                if (os.path.isabs(my_feat_data_fname)):
+                    node_features[ntype_name+'/'+feat_name] = \
+                            torch.from_numpy(np.load(my_feat_data_fname))
+                else:
+                    node_features[ntype_name+'/'+feat_name] = \
+                            torch.from_numpy(np.load(os.path.join(input_dir, my_feat_data_fname)))
+
+                node_feature_tids[ntype_name].append([feat_name, -1, -1])
+
+    '''
+        "node_type" : ["ntype0-name", "ntype1-name", ....], #m node types
+        "num_nodes_per_chunk" : [
+            [a0, a1, ...a<p-1>], #p partitions
+            [b0, b1, ... b<p-1>], 
+            ....
+            [c0, c1, ..., c<p-1>] #no, of node types
+        ],
+
+    The "node_type" points to a list of all the node names present in the graph
+    And "num_nodes_per_chunk" is used to mention no. of nodes present in each of the
+    input nodes files. These node counters are used to compute the type_node_ids as
+    well as global node-ids by using a simple cumulative summation and maitaining an
+    offset counter to store the end of the current.
+
+    '''
 
     #read my nodes for each node type
     node_tids = {}
-    node_data = schema_map["nid"]
-    for ntype_name, ntype_info in node_data.items():
-        v = []
-        node_file_info = ntype_info["data"]
-        for idx in range(len(node_file_info)):
-            v.append((node_file_info[idx][1], node_file_info[idx][2]))
-        node_tids[ntype_name] = v
+    ntype_gnid_offset = {}
+    gnid_offset = 0
+    ntype_names = schema_map[constants.STR_NODE_TYPE]
+    for idx, counts in enumerate(schema_map[constants.STR_NUM_NODES_PER_CHUNK]):
+        ntype_name = ntype_names[idx]
+        ntype_gnid_offset[ntype_name] = gnid_offset 
+        type_nid_start = np.cumsum([0] + counts[:-1])
+        type_nid_end = np.cumsum(counts)
+        type_nid_ranges = list(zip(type_nid_start, type_nid_end))
```

Comment:
this logic appears again. we should have a function to convert it.

Reply:
Done.